geopandas/geoseries.py

```diff
@@ -326,21 +326,26 @@ def notnull(self):
         """Alias for `notna` method. See `notna` for more detail."""
         return self.notna()
 
-    def fillna(self, value=None, method=None, inplace=False,
-               **kwargs):
-        """Fill NA/NaN values with a geometry (empty polygon by default).
-
-        "method" is currently not implemented for pandas <= 0.12.
-        """
+    def fillna(self, value=None, method=None, inplace=False, **kwargs):
+        """ Fill NA/NaN values with a geometry (empty polygon by default) """
+        assert method is None and not inplace
         if value is None:
             value = BaseGeometry()
-        return super(GeoSeries, self).fillna(value=value, method=method,
-                                             inplace=inplace, **kwargs)
+        return GeoSeries(self._geometry_array.fillna(value), index=self.index,
+                         crs=self.crs, name=self.name)
+
+    def dropna(self, method=None, inplace=False, **kwargs):
```

Comment:
I don't think `method` is keyword for dropna in pandas, so probably copied from fillna. Signature for dropna: ` def dropna(self, axis=0, inplace=False)`, but the axis cannot be anything else as 0

Reply:
Fixed.

However given that these aren't supported I am tempted to just not include them in the signature.  The default error message for an extra keyword seems decently informative.

What are your thoughts on this general practice?  My guess is that it is a deviation from normal geopandas development.  I'm happy to go either way.

Reply:
Yes, I am fine with not including those kwargs

(it is there in pandas mainly for dataframe/series compatibility, as many functions are implemented for both with common code, although that is not the case for dropna)